rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_service.hpp

```diff
@@ -0,0 +1,173 @@
+
+  /// Default constructor.
+  /**
+   * The constructor for a LifecycleService is almost never called directly.
+   * Instead, services should be instantiated through the function
+   * rclcpp::create_service().
+   *
+   * \param[in] node_handle NodeBaseInterface pointer that is used in part of
+   * the setup. \param[in] service_name Name of the topic to publish to.
+   * \param[in] any_callback User defined callback to call when a client request
+   * is received. \param[in] service_options options for the subscription.
+   */
+  LifecycleService(
+    std::shared_ptr<rcl_node_t> node_handle, const std::string & service_name,
+    rclcpp::AnyServiceCallback<ServiceT> any_callback, rcl_service_options_t & service_options)
+  : rclcpp::Service<ServiceT>(node_handle, service_name, any_callback, service_options),
+    enabled_(false),
+    should_log_(true),
+    logger_(rclcpp::get_logger("LifecycleService"))
+  {
+  }
+
+  /// Default constructor.
+  /**
+   * The constructor for a LifecycleService is almost never called directly.
+   * Instead, services should be instantiated through the function
+   * rclcpp::create_service().
+   *
+   * \param[in] node_handle NodeBaseInterface pointer that is used in part of
+   * the setup. \param[in] service_handle service handle. \param[in]
+   * any_callback User defined callback to call when a client request is
+   * received.
+   */
+  LifecycleService(
+    std::shared_ptr<rcl_node_t> node_handle, std::shared_ptr<rcl_service_t> service_handle,
+    rclcpp::AnyServiceCallback<ServiceT> any_callback)
+  : rclcpp::Service<ServiceT>(node_handle, service_handle, any_callback),
+    enabled_(false),
+    should_log_(true),
+    logger_(rclcpp::get_logger("LifecycleService"))
+  {
+  }
+
+  /// Default constructor.
+  /**
+   * The constructor for a LifecycleService is almost never called directly.
+   * Instead, services should be instantiated through the function
+   * rclcpp::create_service().
+   *
+   * \param[in] node_handle NodeBaseInterface pointer that is used in part of
+   * the setup. \param[in] service_handle service handle. \param[in]
+   * any_callback User defined callback to call when a client request is
+   * received.
+   */
+  LifecycleService(
+    std::shared_ptr<rcl_node_t> node_handle, rcl_service_t * service_handle,
+    rclcpp::AnyServiceCallback<ServiceT> any_callback)
+  : rclcpp::Service<ServiceT>(node_handle, service_handle, any_callback),
+    enabled_(false),
+    should_log_(true),
+    logger_(rclcpp::get_logger("LifecycleService"))
+  {
+  }
+
+  LifecycleService() = delete;
+
+  virtual ~LifecycleService() {}
+
+  /// add to wait set
+  /**
+   * The function checks whether the communication
+   * was enabled or disabled and forwards the add_to_wait_set
+   * request to the actual rclcpp Service base class
+   */
+  bool add_to_wait_set(rcl_wait_set_t * wait_set) override
+  {
+    if (!enabled_) {
+      log_service_not_enabled();
```

Comment:
I'm don't think that we need to log in the service.
We do have this log in publishers but that's a different situation.

In a publisher, the application code is manually invoking the publish function and if that happens when the entity is not enabled it is likely to be an indication of a bug (e.g. incorrect flow of operations, concurrency, etc).
On the other hand, when it comes to servers, if the user explicitly disables the server then it should expect that requests will not be processed.
The difference with the publisher case is that incoming requests are not an indication of a bug, as they are not something that is manually invoked from the server context.

Moreover, the standard use-case for services/subscription/clients will be that you create them and immediately add to an executor.
This means that the log will immediately trigger regardless of if there are actually incoming requests or not.

Reply:
Make sense, let me remove it.